File: 2022-2023/sourcecode/lab/lab2/ParcelableExampleJava/app/src/main/java/pl/edu/uwr/pum/parcelableexamplejava/Properties.java
package pl.edu.uwr.pum.parcelableexamplejava;

import android.os.Parcel;
import android.os.Parcelable;

public class Properties implements Parcelable {
    private int a;
    private int b;
    private String c;

    public Properties(int a, int b, String c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    protected Properties(Parcel in) {
        a = in.readInt();
        b = in.readInt();
        c = in.readString();
    }

    public static final Creator<Properties> CREATOR = new Creator<Properties>() {
        @Override
        public Properties createFromParcel(Parcel in) {
            return new Properties(in);
        }

        @Override
        public Properties[] newArray(int size) {
            return new Properties[size];
        }
    };

    public int getA() {
        return a;
    }

    public void setA(int a) {
        this.a = a;
    }

    public int getB() {
        return b;
    }

    public void setB(int b) {
        this.b = b;
    }

    public String getC() {
        return c;
    }

    public void setC(String c) {
        this.c = c;
    }

    @Override
    public int describeContents() {
        return hashCode();
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(a);
        parcel.writeInt(b);
        parcel.writeString(c);
    }
}

File: 2022-2023/sourcecode/lab/lab2/ParcelableExampleKotlin/app/src/main/java/pl/edu/uwr/pum/parcelableexamplekotlin/Properties.kt
package pl.edu.uwr.pum.parcelableexamplekotlin

import android.os.Parcel
import android.os.Parcelable

class Properties(val a: Int, val b: Int, val c: String?) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readInt(),
        parcel.readString()
    ) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(a)
        parcel.writeInt(b)
        parcel.writeString(c)
    }

    override fun describeContents(): Int {
        return hashCode()
    }

    companion object CREATOR : Parcelable.Creator<Properties> {
        override fun createFromParcel(parcel: Parcel): Properties {
            return Properties(parcel)
        }

        override fun newArray(size: Int): Array<Properties?> {
            return arrayOfNulls(size)
        }
    }
}